app/src/org/commcare/utils/EncryptionUtils.java

```diff
@@ -171,25 +183,24 @@ public static byte[] generatePassphrase() {
         return result;
     }
 
-    private static String getTransformationString() {
+    public static String getTransformationString(boolean forceRSA) {
         String transformation;
-        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.M) {
-            transformation = String.format("%s/%s/%s", ALGORITHM, BLOCK_MODE, PADDING);
+        if (forceRSA || android.os.Build.VERSION.SDK_INT < android.os.Build.VERSION_CODES.M) {
```

Comment:
The android version check doesn't seem to be meaningful now to me. We can probably just decide the transformation based on `forceRSA` ?

Reply:
Yup, you're right, forceRSA will always be true for API<M. But we get a compile warning about using the ALGORITHM, BLOCK_MODE, and PADDING constants if I remove that check, so I left it there to keep the compiler happy. Would it be better to remove it and ignore/suppress the warning?

Reply:
> Would it be better to remove it and ignore/suppress the warning?

Yeah I would support that since we are already doing this check outside to determine `forceRSA`

Reply:
Done, I removed the extra API check and suppressed the warning.